### PR TITLE
Adding gmm segmentation code

### PR DESCRIPTION
**Pull request by ponl**

### CODE REVIEW

**Comment by ponl** [comment]
@wesleyktatum This class should be ready to go. You just have to instantiate the class and then call `fit_transform` on the material property array.

File: ssts/segmentation/segmentation_gmm.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                data = SegmenterGMM.normalize_by_max(data)
+
+        if self.embedding_dim is not None:
+            self.pca = PCA(n_components=self.embedding_dim)
+            data = self.pca.fit_transform(data)
+
+        self.gmm = GaussianMixture(n_components=self.n_components, covariance_type="full")
+        self.gmm.fit(data)
+
+        return self
```
Comment (dtorrejo) [inline ssts/segmentation/segmentation_gmm.py:87]: Probably don't need this.

File: ssts/segmentation/segmentation_gmm.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        returns a labelled array.
+
+        Args:
+            data (NumPy Array): Material properties array of shape (height, width, n_properties)
+            outliers (NumPy Array): Binary array indicating outliers of shape (height, width)
+
+        Returns:
+            labels (NumPy Array): The segmented array. Each pixel receives
+                a label corresponding to its segment, of shape (height, width).
+        """
```
Comment (dtorrejo) [inline ssts/segmentation/segmentation_gmm.py:102]: Probably add check that len(data.shape) == 3

File: ssts/segmentation/segmentation_gmm.py (hunk 1 above)
Comment (ponl) [reply ssts/segmentation/segmentation_gmm.py:87]: I believe this is what scikit does. It allows you to call
`SegmenterGMM().fit(X)` and get back the fitted segmenter

File: ssts/segmentation/segmentation_gmm.py (hunk 2 above)
Comment (ponl) [reply ssts/segmentation/segmentation_gmm.py:102]: yes, I should check for this

**Pull request merged**